src/commands/install/install.py

```diff
@@ -0,0 +1,48 @@
+from enum import Enum
+from os import path
+from typing import Optional
+from git import InvalidGitRepositoryError, NoSuchPathError, Repo
+from git.objects import Submodule
+from typing_extensions import TypedDict
+from .utils import map_url_to_package_name, verify_repo_url
+
+
+class InstallationErrorEnum(Enum):
+    INCORRECT_URL = "INCORRECT_URL"
+    INVALID_LOCAL_REPOSITORY = "INVALID_LOCAL_REPOSITORY"
+    INVALID_PACKAGE_NAME = "INVALID_PACKAGE_NAME"
+
+
+InstallationResultType = TypedDict(
+    "InstallationResult", {"error": Optional[InstallationErrorEnum]}
+)
+
+
+def install(
```

Comment:
That command does not work for me when i `rm -rf` ./lib - can we make it work like `node-modules`?

Reply:
I'll try to figure something out when implementing the `uninstall` command. Removing lib directory is not enough because submodule information is added in `.git` directory.

Reply:
https://github.com/software-mansion/protostar/issues/6